extension/test/data_test.ts

```diff
@@ -69,4 +69,209 @@ describe('data.ts', function () {
       ).to.include.something.like({ entry: /^ＤＶＤ-ＲＯＭ .*/ });
     });
   });
+
+  describe('makeText', function () {
+    describe('with a non-kanji dict entry', function () {
```

Comment:
having an empty kanji is important but so is having present `data` property. Maybe there's a way to have both.

`with a dict entry consisting of empty kanji but present data properties` or similar...

Reply:
Yeah that's a good point, I'll add that so the test is more explicit about the requirements.

It seems like right now the `kanji` property is used to determine whether a dict entry is a kanji dict entry or a non-kanji/word dict entry. That's kind of confusing because there are a bunch of properties that the kanji dict entry ignores and properties that the non-kanji/word dict entry ignores. So maybe in the future we might want to refactor the code so that these types are more explicit and easier to understand. It might help with readability and it might make impossible types impossible (e.g. it should be impossible for a word dict entry to have a non-empty kanji property). Maybe something like this:
```ts
type DictEntry = KanjiDictEntry | WordDictEntry;
type KanjiDictEntry = {
  kind: 'kanji';
  // properties only used by kanji dict entry
};
type WordDictEntry = {
  kind: 'word';
  // properties only used by word dict entry
};
```
Then in the code which wants to deal with a dict entry, we could maybe get exhaustive type checking which is more explicit, something like this:
```ts
switch (dictEntry.kind) {
  case 'kanji':
    return // do things with kanji entry
  case 'word':
    return // do things with word entry
}
```
Not sure if this is a good idea, but just brainstorming ways to make the code more explicit and easier to understand in the future.

Reply:
Something like that seems good I agree.

Using Discriinated Unions like you describe seems reasonable! Another idea would be to consider if any functions actually need to operate on both kinds of dict entries. For example, `makeTextForKanji` and `makeTextForWord` as separate functions could be better depending how exactly they're called. 

In general, branching early is better than later (makes code easy to reason about) but it really depends on how much shared code there is!

Definitely worth thinking about. There's a ton of legacy code that I inherited so no need to hold back. :D